scope

```diff
@@ -136,48 +147,53 @@ container_ip() {
 
 case "$COMMAND" in
 
-	launch)
-		check_not_running $CONTAINER_NAME $SCOPE_IMAGE
-
-		# If WeaveDNS is running, we want to automatically tell the scope
-		# image to use weave dns.  We can't use --dns with --net=host, so we have to hack it.
-		if dns_running; then
-			docker_bridge_ip
-			WEAVESCOPE_DNS_ARGS="$WEAVESCOPE_DNS_ARGS --dns $DOCKER_BRIDGE_IP --searchpath $DOMAINNAME"
-		fi
-
-		CONTAINER=$(docker run --privileged -d --name=$CONTAINER_NAME --net=host --pid=host \
-			-v /var/run/docker.sock:/var/run/docker.sock \
-			$WEAVESCOPE_DOCKER_ARGS $SCOPE_IMAGE $WEAVESCOPE_DNS_ARGS "$@")
-
-		IP_ADDRS=$(docker run --net=host gliderlabs/alpine /bin/sh -c "$IP_ADDR_CMD")
-		if dns_running; then
-			if [ -z "$IP_ADDRS" ]; then
-				echo "Could not determine local IP address; Weave DNS integration will not work correctly."
-				exit 1
-			fi
-			tell_dns_fqdn PUT $CONTAINER $FQDN $IP_ADDRS
-		fi
-
-		echo $CONTAINER
-
-		echo "Weave Scope is reachable at the following URL(s):" >&2
-		for ip in $IP_ADDRS; do
-			echo "  * http://$ip:4040/" >&2
-		done
-		;;
-
-	stop)
-		[ $# -eq 0 ] || usage
-		if ! docker stop $CONTAINER_NAME >/dev/null 2>&1 ; then
-			echo "Weave Scope is not running." >&2
-		fi
-		docker rm -f $CONTAINER_NAME >/dev/null 2>&1 || true
-		;;
-
-	*)
-		echo "Unknown scope command '$COMMAND'" >&2
-		usage
-		;;
+    launch)
+        check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE
+
+        # If Weave is running, we want to expose a Weave IP to the host
+        # network namespace, so Scope can use it.
+        weave_expose
+
+        # If WeaveDNS is running, we want to automatically tell the scope
+        # image to use weave dns.  We can't use --dns with --net=host, so we
+        # have to hack it.
+        if is_running $DNS_CONTAINER_NAME; then
```

Comment:
I'm just worried that scope launch will stop working when we ship 1.1  OTOH we probably shouldn't make assumptions about what 1.1 will look like.  And we can release a new scope version to deal with 1.1 super quick.

So yeah, not doing anything now might be best idea.